src/components/QrScanner/QrScannerScreenContainer.tsx

```diff
@@ -25,15 +25,31 @@ export const QrScannerScreenContainer: FunctionComponent<QrScannerScreenContaine
     navigation.navigate("ScannedDocumentScreen", { document });
   };
   const onQrData = async (data: string): Promise<void> => {
+    if (scanningDisabled) {
+      return;
+    }
     setScanningDisabled(true);
     try {
       await processQr(data, { onDocumentStore, onDocumentView });
     } catch (e) {
+      setScanningDisabled(false);
```

Comment:
Think this needs to be outside of the try catch, if not after if there is a case where the processQr does not lead to this screen being unmounted, the `scanningDisabled` will still be true

Reply:
Hmm, so you're saying that `processQr` succeeds but doesn't cause a navigation?

Reply:
Yep!

Reply:
As spoken, given that `processQr` will either cause a navigation or throw an Error, we'll leave this in. The other issue with moving it out was that it gets called after the screen blurs, which makes the qr scanning work when we don't want it to.